tools/tpm2_nvreadlock.c

```diff
@@ -14,16 +14,22 @@ struct tpm_nvreadlock_ctx {
         tpm2_loaded_object object;
     } auth_hierarchy;
 
+    bool is_auth_hierarchy_specified;
```

Comment:
no need for this bool, just check if path is NULL

Reply:
Done, I also changed this in `tpm2_nvincrement`, `tpm2_nvread` and `tpm2_nvwrite` to keep the code consistent everywhere.